### PR TITLE
feat(e-commerce): e-commerce screens add 

### PR DESCRIPTION
**Pull request by 32penkin**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/kittenTricks/blob/master/CONTRIBUTING.md) guide.

 #### Short description of what this resolves:

### CODE REVIEW

File: src/components/ecommerce/ecommerceInput.component.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { ThemedComponentProps } from '@kitten/theme';
+import {
+  Input,
+  InputProps,
+} from '../common/input.component';
+
+interface ComponentProps {
+  formatter: (value: string, stateValue: string) => string;
+  validator: (value: string) => boolean;
```
Comment (artyorsh) [inline src/components/ecommerce/ecommerceInput.component.tsx]: can we use `ValidateInput` with it's regex prop for this purpose?

File: src/components/ecommerce/formatters.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      formatted += '/';
+    } else {
+      formatted = formatted[0];
+    }
+  }
+  return formatted;
+};
+
+export const CvvFormatter = (value: string): string => {
+  return value;
```
Comment (artyorsh) [inline src/components/ecommerce/formatters.ts:26]: very nice :)

File: src/components/ecommerce/paymentCard.component.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    lineHeight: 42,
+  },
+  numberLabel: {
+    fontFamily: 'opensans-bold',
+    fontSize: 18,
+    lineHeight: 24,
+    marginBottom: 26,
+  },
+  bottomLabel: {
+    fontWeight: 'normal',
```
Comment (artyorsh) [inline src/components/ecommerce/paymentCard.component.tsx]: this is default value, isn't it?

File: src/components/ecommerce/validators.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+export const ExpirationDateValidator = (value: string): boolean => {
+  return value.length === 5;
+};
+
+export const CvvValidator = (value: string): boolean => {
+  return value.length === 3;
+};
+
+export const CardholderNameValidator = (value: string): boolean => {
+  return /^$|^[a-zA-ZčČćĆđĐšŠžŽ-]+ [a-zA-ZčČćĆđĐšŠžŽ-]+$/.test(value) && value.length !== 0;
```
Comment (artyorsh) [inline src/components/ecommerce/validators.ts:14]: prefer refactoring to use existing regex for name

File: src/containers/layouts/ecommerce/addNewCard/addNewCard.component.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+          <Text style={themedStyle.inputLabel}>CARD NUMBER</Text>
+          <EcommerceInput
+            style={[themedStyle.input, themedStyle.commonMargin]}
+            value={cardNumber}
+            validator={CardNumberValidator}
+            formatter={CardNumberFormatter}
+            placeholder='0000 0000 0000 0000'
+            maxLength={19}
+            keyboardType='numeric'
+            onChangeText={this.onCardNumberChange}/>
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/addNewCard/addNewCard.component.tsx]: jsx-closing-bracket-location

File: src/containers/layouts/ecommerce/productDetails/byu.modal.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  public render(): React.ReactNode {
+    const {
+      themedStyle,
+      shoppingItemsCount,
+      product,
+      actionItems,
+    } = this.props;
+
+    return (
+      [
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/productDetails/byu.modal.tsx:61]: why is this an array?

File: src/containers/layouts/ecommerce/productDetails/byu.modal.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  container: {
+    zIndex: 1,
+    backgroundColor: theme['color-white'],
+    justifyContent: 'space-between',
+    paddingTop: 24,
+    paddingBottom: 16,
+    width: 343,
+    height: 276,
+    borderRadius: 12,
+    top: (height - 276) / 2,
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/productDetails/byu.modal.tsx:108]: this looks crazy, is this a framework issue?

File: src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+  RadioProps,
+  Button,
+  Input,
+} from '@kitten/ui';
+import { Text } from '@src/components/common';
+import {
+  Product as ProductModel,
+  Comment as CommentModel,
+} from '@src/core/model';
+import { CommentsList1 } from '@src/containers/layouts/messaging/commentsList1/commentsList1.component';
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx]: should be `@src/containers/layouts/messaging` or moved to `common` or `messaging` components

File: src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+    return (
+      <Radio
+        style={themedStyle.radioItem}
+        key={index}
+        text={color}
+      />
+    );
+  };
+
+  private renderInfoContainer = (): React.ReactElement<ViewProps> => {
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx:91]: prefer refactoring `render` methods to components even if it's used once

File: src/containers/layouts/ecommerce/productsList/productsList.component.tsx [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+        onAddToBucket={this.onAddToBucketPress}
+      />
+    );
+  };
+
+  private renderProducts = (): React.ReactElement<ViewProps> => {
+    const { themedStyle } = this.props;
+
+    return (
+      <View style={themedStyle.productsList}>
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/productsList/productsList.component.tsx]: The same. Looks much better if it is separated component

File: src/containers/layouts/ecommerce/productsList/productsList.component.tsx [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+      </View>
+    );
+  };
+
+  public render(): React.ReactNode {
+    const { themedStyle } = this.props;
+    const tabs: string[] = this.getComponentsTabs();
+
+    return (
+      <View style={themedStyle.container}>
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/productsList/productsList.component.tsx:122]: Could this be achieved with `contentContainerStyle` prop?

File: src/containers/layouts/ecommerce/shoppingCart/shoppingCart.component.tsx [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+import {
+  List,
+  Button,
+} from '@kitten/ui';
+import { Text } from '@src/components/common';
+import { CartProduct as CartProductModel } from './cartProduct.model';
+import {
+  CartProduct,
+  CartProductProps,
+} from '@src/components/ecommerce/cartProduct.component';
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/shoppingCart/shoppingCart.component.tsx]: create indexes for each category of components, so you'll achieve `@src/components/ecommerce`

File: src/containers/layouts/ecommerce/shoppingCart/shoppingCart.component.tsx [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+
+  public componentWillMount(): void {
+    this.currency = this.props.products[0].currency;
+  }
+
+  private calculateTotalPrice = (): number => {
+    const { products } = this.props;
+
+    return products
+      .reduce((acc: number, item: CartProductModel) => acc + item.count * item.cost, 0);
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/shoppingCart/shoppingCart.component.tsx]: do not avoid closure brackets

File: src/containers/layouts/ecommerce/shoppingCart/shoppingCart.container.tsx [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+interface State {
+  cartProducts: CartProduct[];
+}
+
+export class ShoppingCartContainer extends React.Component<NavigationScreenProps, State> {
+
+  private prepareProducts = (): CartProduct[] => {
+    return [products[0], products[1]]
+      .map((item: Product) => ({
+        ...item,
```
Comment (artyorsh) [inline src/containers/layouts/ecommerce/shoppingCart/shoppingCart.container.tsx]: much better if such cases are formatted either mapping function or has `return` statement, which is omitted here

**Review by artyorsh — APPROVED** [review]

File: src/components/ecommerce/ecommerceInput.component.tsx (hunk 1 above)
Comment (32penkin) [reply src/components/ecommerce/ecommerceInput.component.tsx]: Don't like this input at all. It forces me to create another one statefull component.

File: src/components/ecommerce/formatters.ts (hunk 2 above)
Comment (32penkin) [reply src/components/ecommerce/formatters.ts:26]: Just a code consistency, all as you like :)

File: src/components/ecommerce/paymentCard.component.tsx (hunk 3 above)
Comment (32penkin) [reply src/components/ecommerce/paymentCard.component.tsx]: Sure. Refactored.

File: src/components/ecommerce/validators.ts (hunk 4 above)
Comment (32penkin) [reply src/components/ecommerce/validators.ts:14]: I need exactly 2 words with 1 space between them.

File: src/containers/layouts/ecommerce/addNewCard/addNewCard.component.tsx (hunk 5 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/addNewCard/addNewCard.component.tsx]: Refactored

File: src/containers/layouts/ecommerce/productDetails/byu.modal.tsx (hunk 6 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/productDetails/byu.modal.tsx:61]: Just backdrop case. There is not yet implemented customization backdrop in the RNUK

File: src/containers/layouts/ecommerce/productDetails/byu.modal.tsx (hunk 7 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/productDetails/byu.modal.tsx:108]: This is not an issue, just one of the options of placing 'absolute' View with strict size 'screen-centered'. We allow the user to do this for flexibility.

File: src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx (hunk 8 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx]: Refactored.

File: src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx (hunk 9 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/productDetails/productDetails.component.tsx:91]: I don't see any sense in that. The reason of creating the separate component - only when it'll be reused multiple times.

File: src/containers/layouts/ecommerce/productsList/productsList.component.tsx (hunk 10 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/productsList/productsList.component.tsx]: I don't see any sense in that. The reason of creating the separate component - only when it'll be reused multiple times.

File: src/containers/layouts/ecommerce/shoppingCart/shoppingCart.component.tsx (hunk 12 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/shoppingCart/shoppingCart.component.tsx]: Done.

File: src/containers/layouts/ecommerce/shoppingCart/shoppingCart.component.tsx (hunk 13 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/shoppingCart/shoppingCart.component.tsx]: Done.

File: src/containers/layouts/ecommerce/shoppingCart/shoppingCart.container.tsx (hunk 14 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/shoppingCart/shoppingCart.container.tsx]: Done

File: src/containers/layouts/ecommerce/productsList/productsList.component.tsx (hunk 11 above)
Comment (32penkin) [reply src/containers/layouts/ecommerce/productsList/productsList.component.tsx:122]: Refactored

**Review by artyorsh — APPROVED** [review]
Can we merge #102 first? This will allow using common `ValidationInput`

**Pull request merged**